execution_engine/src/storage/global_state/lmdb.rs

```diff
@@ -94,14 +94,33 @@ impl LmdbGlobalState {
         prestate_hash: Digest,
         stored_values: HashMap<Key, StoredValue>,
     ) -> Result<Digest, error::Error> {
-        put_stored_values::<LmdbEnvironment, LmdbTrieStore, error::Error>(
-            &self.environment,
-            &self.trie_store,
+        let scratch_trie = self.get_scratch_store();
+        let new_state_root = put_stored_values::<_, _, error::Error>(
+            &scratch_trie,
+            &scratch_trie,
```

Comment:
Is it intentional that we use the same ScratchStore twice?

Reply:
Yep, we need to emulate a transaction to make use of the generic traits `TransactionSource`, `Readable`, `Writable`, and related `write` `scan` and `read` functions. I think this area is ripe for refactoring, in particular once rocksdb lands, but is entirely out of scope for this changeset.